pkg/playback/http_playback.go

```diff
@@ -431,28 +424,40 @@ func (rr *Server) ejectCassette() error {
 		return fmt.Errorf("tried to eject when no cassette is loaded")
 	}
 
-	switch rr.mode {
-	case Record:
+	isRecording, err := rr.isRecording()
+
+	if err != nil {
+		// We should never get here, since all mutations of rr.mode should have already validated the mode value.
+		rr.log.Fatal(err.Error())
+		return err
+	}
+
+	if isRecording {
 		err := rr.httpRecorder.recorder.saveAndClose()
 		if err != nil {
 			rr.cassetteLoaded = false // if an error occurs, best to reset to an blank state so a new cassette can be loaded
 			return fmt.Errorf("unexpected error when writing cassette. It may have failed to write properly: %w", err)
 		}
-	case Replay:
-		// nothing
-	case Auto:
-		if rr.isRecordingInAutoMode {
-			err := rr.httpRecorder.recorder.saveAndClose()
-			if err != nil {
-				rr.cassetteLoaded = false // if an error occurs, best to reset to an blank state so a new cassette can be loaded
-				return fmt.Errorf("unexpected error when writing cassette. It may have failed to write properly: %w", err)
-			}
-		}
 	}
+	// ejecting is a no-op internally when in replay mode
+
 	rr.cassetteLoaded = false
 	return nil
 }
 
+func (rr *Server) isRecording() (isRecording bool, err error) {
+	switch rr.mode {
+	case Record:
+		return true, nil
+	case Replay:
+		return false, nil
+	case Auto:
+		return rr.isRecordingInAutoMode, nil
+	default: //throw
+		return false, fmt.Errorf("Unexpected mode \"%v\"in playback server - this likely indicates a bug in the implementation. Please try restarting the server", rr.mode)
```

Comment:
It would be better to crash here, I think. Then you wouldn't need to pipe around the extra `err`, and it doesn't really help the UX to continue.

Reply:
Good point! 990af9d10cc2e01d539f4df3cfa58d054609a61b